engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -263,6 +263,47 @@ namespace dmGameObject
         return regist->m_DefaultInputStackCapacity;
     }
 
+    void AddDynamicResourceHash(HCollection hcollection, dmhash_t resource_hash)
+    {
+        Collection* collection = hcollection->m_Collection;
+        dmMutex::Lock(collection->m_Mutex);
+        if (collection->m_DynamicResources.Remaining() == 0)
+        {
+            collection->m_DynamicResources.OffsetCapacity(1);
+        }
+        collection->m_DynamicResources.Push(resource_hash);
+        dmMutex::Unlock(collection->m_Mutex);
+    }
+
+    void RemoveDynamicResourceHash(HCollection hcollection, dmhash_t resource_hash)
+    {
+        Collection* collection = hcollection->m_Collection;
+        dmMutex::Lock(collection->m_Mutex);
+        for (int i = 0; i < collection->m_DynamicResources.Size(); ++i)
+        {
+            if (collection->m_DynamicResources[i] == resource_hash)
+            {
+                collection->m_DynamicResources.EraseSwap(i);
+            }
+        }
+        dmMutex::Unlock(collection->m_Mutex);
+    }
+
+    static void ReleaseDynamicResources(Collection* collection)
+    {
+        dmMutex::Lock(collection->m_Mutex);
+        for (int i = 0; i < collection->m_DynamicResources.Size(); ++i)
+        {
+            dmResource::SResourceDescriptor* rd = dmResource::FindByHash(collection->m_Factory, collection->m_DynamicResources[i]);
+            assert(rd);
+            void* resource = dmResource::GetResource(rd);
+            dmResource::Release(collection->m_Factory, resource);
```

Comment:
This call expects that the factory that created the resource is the same that is now releasing it, which should be the case but I don't know if we can trust that 100%? The resource script module uses a global factory for creating the resource, but can we trust that it's the same that's releasing it?

Reply:
The reason we have a pointer to the factory, is to avoid using singletons like "GetGlobalFactory()"